core/federated/RTI/rti_remote.c

```diff
@@ -1063,32 +1064,52 @@ void handle_address_ad(uint16_t federate_id) {
 }
 
 /**
- * Send to the start time to the federate my_fed.
- * This function assumes the caller does not hold the mutex.
+ * Send the start time and tag to the federate my_fed.
  *
- * If it is the startup phase, the start_time will be the maximum received timestamps
- * plus an offset. The federate will then receive identical federation_start_time
- * and federate_start_tag.time (the federate_start_tag.microstep will be 0).
- * If, however, the startup phase is passed, the federate will receive different
- * values than stated above.
+ * During the startup phase, the start_time is calculated as the maximum received timestamp, plus an offset.
+ * The federate will then receive identical values for federation_start_time` and `federate_start_tag.time` (with
+ * `federate_start_tag.microstep` set to 0). After the startup phase, the federate will receive different values for
+ * these parameters.
  *
- * This will also notify federates downstream of my_fed that this federate is now
- * connected.  This is important when there are zero-delay cycles.
+ * Before sending the start time and tag, this function performs the following actions:
+ *  - If my_fed is transient, notify federates downstream of its connection, ensuring proper handling of zero-delay
+ *    cycles.
```

Comment:
There is no need for this to be done before the messages to `my_fed`.  Note that there is no ordering semantics between messages sent to different federates.  I suggest moving this code to _after_ sending sending the start tag so that the notification does not occur if sending the start tag fails.

Reply:
Yeah, makes sense. Added to the upcoming commit.